std-bits/base/src/main/java/org/enso/base/Array_Builder.java

```diff
@@ -1,13 +1,17 @@
 package org.enso.base;
 
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 public class Array_Builder<T> {
-  private final List<T> builder;
+  private static final Object[] EMPTY_ARRAY = new Object[0];
+  private final int capacity;
+  private int size;
+  private Object primitiveArray;
+  private Object[] objectArray;
 
   private Array_Builder(int capacity) {
-    this.builder = new ArrayList<>(capacity);
+    this.capacity = Math.max(2, capacity);
```

Comment:
This lower bound will make us always copy when `capacity == 1` and we indeed want only a single element. Right? Is there any reason for this capacity lower bound?

Reply:
Fixed in https://github.com/enso-org/enso/pull/3809/commits/9aa1cf8b7ebfa55e4a7b3b5805e042175133fbf7 to allow capacity `1`.